[Flang] Fix symbol name clash when dummy procedure name is the same as common-block-name

ceseo opened the pull request:
Dummy procedures in interface blocks are not external procedures that need to be linked. Do not externalize those.

Fixes #122822

clementval commented on flang/lib/Optimizer/Transforms/ExternalNameConversion.cpp:
```diff
@@ -82,6 +82,48 @@ void ExternalNameConversionPass::runOnOperation() {
+                }
+              }
+            }
+            if (!isDirectlyCalled) {
```
No brace on simple if

clementval commented on flang/lib/Optimizer/Transforms/ExternalNameConversion.cpp:
```diff
@@ -82,6 +82,48 @@ void ExternalNameConversionPass::runOnOperation() {
+            auto uses = funcOp.getSymbolUses(mod);
+            if (uses.has_value()) {
+              for (auto use : *uses) {
+                auto *user = use.getUser();
```
Spell out auto

clementval commented on flang/lib/Optimizer/Transforms/ExternalNameConversion.cpp:
```diff
@@ -82,6 +82,48 @@ void ExternalNameConversionPass::runOnOperation() {
+          // specifications)
+          if (hasConflictingCommonBlock) {
+            bool isDirectlyCalled = false;
+            auto uses = funcOp.getSymbolUses(mod);
```
Spell out auto

clementval commented on flang/lib/Optimizer/Transforms/ExternalNameConversion.cpp:
```diff
@@ -82,6 +82,48 @@ void ExternalNameConversionPass::runOnOperation() {
+      // block and get its mangled name.
+      if (auto funcOp = llvm::dyn_cast<mlir::func::FuncOp>(funcOrGlobal)) {
+        if (funcOp.isPrivate()) {
+          auto mangledName =
```
Spell out auto

ceseo commented on flang/lib/Optimizer/Transforms/ExternalNameConversion.cpp:
```diff
@@ -82,6 +82,46 @@ void ExternalNameConversionPass::runOnOperation() {
+          bool hasConflictingCommonBlock = false;
+
+          // Check if any existing global has the same mangled name.
+          mlir::SymbolTable symbolTable(mod);
```
OK, I'll take a look. Thanks!

ceseo commented:
@clementval any other comments about this? Thanks!

clementval reviewed (commented):
Overall it looks good. I'm just thinking if your case requires to keep the symbol table up to date with the other renaming happening in this pass.

ceseo commented:
> Overall it looks good. I'm just thinking if your case requires to keep the symbol table up to date with the other renaming happening in this pass.

OK, I'll rebase against the current master since it's been a while this is under review. If you have any suggestions for a test that might expose potential issues, please let me know.

ceseo committed:
[Flang] Fix symbol name clash when dummy procedure name is the same as common-block-name

Dummy procedures in interface blocks are not external procedures that need to be linked. Do not externalize those.

Fixes #122822

clementval commented:
> > > > Overall it looks good. I'm just thinking if your case requires to keep the symbol table up to date with the other renaming happening in this pass.
> > > 
> > > 
> > > OK, I'll rebase against the current master since it's been a while this is under review. If you have any suggestions for a test that might expose potential issues, please let me know.
> > 
> > 
> > Your rebase likely went wrong.
> 
> Sorry about that. Git was configured to merge by default.

You can fix it and force push
